.circleci/config.yml

```diff
@@ -13,43 +13,35 @@ commands:
       - run: brew install protobuf cmake ccache libtool libspatialite pkg-config luajit curl wget czmq lz4 spatialite-tools unzip
       - run: pip3 install requests shapely conan
       - run: git clone https://github.com/kevinkreiser/prime_server --recurse-submodules && cd prime_server && ./autogen.sh && ./configure && make -j8 && make install
+  install_linux_dependencies:
+    steps:
+      - run: sudo bash ./scripts/install-linux-deps.sh
 
 jobs:
   build-docker-images:
-    docker:
-      - image: docker:18.06.3-ce-git
+    machine:
+      image: ubuntu-2004:2022.07.1
     steps:
-      - setup_remote_docker
       - checkout
       - run:
-          name: Build Valhalla Docker Images
+          name: Build and push Valhalla Docker image
           command: |
-            docker build --pull -f ./docker/Dockerfile-build --tag valhalla/valhalla:build-latest ./docker
             if [[ "${CIRCLE_BRANCH}" == "master" ]] || [[ ! -z "${CIRCLE_TAG}" ]]; then
-              docker build -f ./docker/Dockerfile-run --tag valhalla/valhalla:run-latest .
-            fi
-      - run:
-          name: Push Valhalla Docker Images
-          command: |
-            if [ "${CIRCLE_BRANCH}" == "master" ]; then
-              echo "$DOCKERHUB_PASS" | docker login -u "$DOCKERHUB_USERNAME" --password-stdin
-              docker push valhalla/valhalla:build-latest
-              docker push valhalla/valhalla:run-latest
-            elif [[ ! -z "${CIRCLE_TAG}" ]]; then
+              BUILD_TAG="${CIRCLE_TAG:-latest}"
               echo "$DOCKERHUB_PASS" | docker login -u "$DOCKERHUB_USERNAME" --password-stdin
-              docker tag valhalla/valhalla:build-latest valhalla/valhalla:build-${CIRCLE_TAG}
-              docker push valhalla/valhalla:build-${CIRCLE_TAG}
-              docker tag valhalla/valhalla:run-latest valhalla/valhalla:run-${CIRCLE_TAG}
-              docker push valhalla/valhalla:run-${CIRCLE_TAG}
+              docker build --tag valhalla/valhalla:run-${BUILD_TAG} .
+              docker push valhalla/valhalla:run-${BUILD_TAG}
             fi
 
   lint-build-debug:
-    docker:
-      - image: valhalla/valhalla:build-latest
+    machine:
+      image: ubuntu-2004:2022.07.1
     resource_class: xlarge
     steps:
       - checkout
+      - install_linux_dependencies
       - run: |
+            python3 -m pip install requests
```

Comment:
confusingly, circleci doesn't use the same python version as installed by our script.. makes no sense and is very annoying